scargo/commands/check.py

```diff
@@ -189,29 +189,46 @@ def check(self) -> None:
             return
         super().check()
 
+    def __get_copyright_lines(self) -> List[str]:
+        return ["//\n"] + [f"// {el}\n" for el in self.copyright_desc.split("\n")] + ["//\n"]
+
     def check_file(self, file_path: Path) -> CheckResult:
+        copyright_lines = self.__get_copyright_lines()
+
         with open(file_path, encoding="utf-8") as file:
-            for line in file.readlines():
-                if self.copyright_desc in line:
-                    return CheckResult(problems_found=0)
-                if "copyright" in line.lower():
-                    logger.warning(
-                        "Incorrect and not excluded copyright in %s", file_path
-                    )
-                    return CheckResult(problems_found=1, fix=False)
+            file_lines = file.readlines()
+
+            for line in file_lines:
+                if line == copyright_lines[0]:
+                    if all(line_from_file == line_from_copyrights for line_from_file, line_from_copyrights in
+                           zip(file_lines[file_lines.index(line):], copyright_lines)):
+                        return CheckResult(problems_found=0)
+
+                    file_lines = file_lines[file_lines.index(line) + 1:]
```

Comment:
Changing a list while it's being iterated is not recommended and may have unforeseen consequences. And it does not seem that you really need it here.
As far as I can tell, you're doing it so that you can get the line index with `list.index()` which is not a good way, it's better to change `for line in file_lines:` to something like `for line_no, line in enumerate(file_lines)` and you get the current line number right there.

Reply:
Changed according to solution proposal.